Tutorials/BasicCameraExample/BasicCameraExample.cs

```diff
@@ -0,0 +1,454 @@
+        modelVelocity += modelVelocityAdd;
+
+        GamePad.SetVibration(PlayerIndex.One, InputManager.RightTriggerValue,
+            InputManager.RightTriggerValue);
+
+        // Set some audio based on whether we're pressing a trigger.
+        if ((InputManager.RightTriggerValue > 0) || InputManager.IsKeyDown(Keys.Up))
+        {
+            if (engineSound.State == SoundState.Stopped)
+            {
+                engineSound = engineSoundEffect.CreateInstance();
+                engineSound.IsLooped = true;
+                engineSound.Play();
+            }
+            else if (engineSound.State == SoundState.Paused)
+            {
+                engineSound.Resume();
+            }
+        }
+        else
+        {
+            if (engineSound.State == SoundState.Playing)
+            {
+                engineSound.Pause();
+            }
+        }
+
+        // In case you get lost, press A to warp back to the center.
+        if (InputManager.IsButtonPressed(Buttons.A) || InputManager.IsKeyPressed(Keys.Space))
+        {
+            modelPosition = new Vector3(0.0f, 350.0f, 0.0f);
+            modelVelocity = Vector3.Zero;
+            modelRotation = 0.0f;
+
+            // Make a sound when we warp.
+            hyperspaceSoundEffect.Play();
+        }
+
+        // Toggle the state of the camera.
+        if (InputManager.IsButtonPressed(Buttons.LeftShoulder) || InputManager.IsKeyPressed(Keys.Tab))
+        {
+            currentCameraMode++;
+            currentCameraMode = (CameraMode)((int)currentCameraMode % 6);
+        }
+
+        // Pressing the A button or key toggles the spring behavior on and off
+        if (InputManager.IsButtonPressed(Buttons.A) || InputManager.IsKeyPressed(Keys.A))
+        {
+            cameraSpringEnabled = !cameraSpringEnabled;
+        }
+    }
+
+    #region Camera Modes
+    /// <summary>
+    /// The following methods update the camera view matrix based on the current camera mode.
+    /// </summary>
+
+    /// <summary>
+    /// Helper method to update the current camera view matrix.
+    /// </summary>
+    void UpdateCameraView(Vector3 aCameraPosition, Vector3 aCameraTarget)
+    {
+        currentCameraView = Matrix.CreateLookAt(aCameraPosition, aCameraTarget, Vector3.Up);
+    }
+
+    void UpdateFixedCamera()
+    {
+        // Fixed view, the camera is always in the same position and looking the same way, no updates.
+        UpdateCameraView(cameraFixedPosition, Vector3.Zero);
+    }
+
+    void UpdateTrackingCamera()
+    {
+        // Tracking view, the camera is always in the same position but changes the view matrix to "look" towards a target.
+
+        // Create a vector pointing the direction the camera is facing.  
+        Vector3 transformedReference = Vector3.Transform(modelPosition, Matrix.Identity);
+
+        // Calculate the position the camera is looking at.
+        Vector3 cameraLookat = transformedReference + modelPosition;
```

Comment:
As the `transformedReference` is just `modelPosition`, this is basically doing `cameraLookat = 2 * modelPosition` which is incorrect. The `modelPosition` is all that is needed here. Currently this is why the ship can go off-screen when passing under the camera.

Reply:
You are indeed correct @squarebananas , nice find.  I wonder why the original XNA tutorial did this,

Method simply updated to:

```csharp
    void UpdateTrackingCamera()
    {
        // Tracking view, the camera is always in the same position but changes the view matrix to "look" towards a target.

        // Set up our world matrix, view matrix and projection matrix.
        UpdateCameraView(cameraFixedPosition, modelPosition);
    }
```

No need for any transformations, just look at the model.